test/correctness/deferred_loop_level.cpp

```diff
@@ -0,0 +1,128 @@
+#include "Halide.h"
+
+using namespace Halide;
+using namespace Halide::Internal;
+
+class CheckScheduleParams : public IRVisitor {
+public:
+    CheckScheduleParams(const std::string &inner_loop_level,
+                        const std::string &outer_loop_level) :
+        inner_loop_level(inner_loop_level), outer_loop_level(outer_loop_level) {}
+
+private:
+    using IRVisitor::visit;
+
+    const std::string inner_loop_level, outer_loop_level;
+    std::string inside_for_loop;
+
+    void visit(const For *op) override {
+        std::string old_for_loop = inside_for_loop;
+        inside_for_loop = op->name;
+        IRVisitor::visit(op);
+        inside_for_loop = old_for_loop;
+    }
+
+    void visit(const Min *op) override {
+        IRVisitor::visit(op);
+        _halide_user_assert(inside_for_loop == inner_loop_level);
```

Comment:
I think I see what the test is trying to do... it uses min/max as unique operations to identify where things got scheduled. I'm nervous about this because min and max get generated during lowering, so this test might be fragile. I can see a few options:

- Use something not generated during lowering, e.g. sin, cos.
- Use floats, and check that the type of the min/max is float.
- I think it's sufficient to simply check the stores (which you are already doing).

Reply:
Yeah, good point. I'll revise it, and also add comments to make the tricks more explicit.